text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/ContextImpl.java

```diff
@@ -138,19 +138,34 @@ public UnaryOperator<String> preProcessor() {
 
   @Override
   public @NotNull Component deserialize(final @NotNull String message) {
-    return this.miniMessage.deserialize(requireNonNull(message, "message"), this.tagResolver);
+    requireNonNull(message, "message");
+    if (this.target != null) {
+      return this.miniMessage.deserialize(message, this.target, this.tagResolver);
+    } else {
+      return this.miniMessage.deserialize(message, this.tagResolver);
+    }
```

Comment:
silly thing but this block is repeated in all the overloads, can you break it out into its own method?